examples/pytorch/catalyst_simple.py

```diff
@@ -97,15 +82,23 @@ def objective(trial):
         logdir=logdir,
         num_epochs=num_epochs,
         verbose=True,
-        callbacks=[
-            AccuracyCallback(),
-            CatalystPruningCallback(
-                trial, metric="accuracy01"
-            ),  # top-1 accuracy as metric for pruning
-        ],
+        callbacks={
+            # top-1 accuracy as metric for pruning
+            "optuna": CatalystPruningCallback(
```

Comment:
Of course we can use OptunaPruningCallback and CatalystPruningCallback interchangeably.

Reply:
Thank you for your suggestion. That's reasonable to me because `CatalystPruningCallback` is deprecated by this PR. If we replace the callback, I want a comment to explain why `OptunaPruningCallback` is used instead of `CatalystPruningCallback`.

Reply:
OK. So let me merge this PR and send a follow-up one to use `OptunaPruningCallback` (and explain why it is used).